client/src/store/reducer.ts

```diff
@@ -268,6 +359,100 @@ function onQuestionInputChanged(
   });
 }
 
+function onQuestionAnswered(
+  state: State,
+  action: QuestionAnsweredAction
+): State {
+  // NOTE: about answerFeedbackId
+  // It seems like the answerFeedbackId should be
+  // associated to the chat message
+  const response = action.mentor;
+  const mentor: MentorState = {
+    ...state.mentorsById[response.mentor],
+    // we need chat messages to live up here
+    answer_id: response.answerId,
+    answer_text: response.answerText,
+    answer_media: response.answerMedia,
+    answerReceivedAt: new Date(Date.now()),
+    answerFeedbackId: response.answerFeedbackId,
+    classifier: response.answerClassifier,
+    confidence: response.answerConfidence,
+    is_off_topic: response.answerIsOffTopic,
+    question: response.question,
+    response_time: response.answerResponseTimeSecs,
+    status: MentorQuestionStatus.READY,
+  };
+  const history = mentor.topic_questions.length - 1;
+  if (!mentor.topic_questions[history].questions.includes(response.question)) {
+    mentor.topic_questions[history].questions.push(response.question);
+  }
+
+  return {
+    ...state,
+    chat: {
+      ...state.chat,
+      messages: [
+        ...state.chat.messages,
+        {
+          name: "",
+          color: "",
+          mentorId: action.mentor.mentor,
+          isUser: false,
+          text: action.mentor.answerText,
+          feedback: Feedback.NONE,
+          feedbackId: action.mentor.answerFeedbackId,
+          isFeedbackSendInProgress: false,
+          visibility: true,
+          clicked: false,
+        },
+      ],
+    },
+    isIdle: false,
+    mentorsById: {
+      ...state.mentorsById,
+      [response.mentor]: mentor,
+    },
+  };
+}
+
+function onQuestionClick(state: State, action: visibilityAnswerAction): State {
+  return {
+    ...state,
+    chat: {
+      ...state.chat,
+      messages: state.chat.messages.map((m, i) => {
+        return action.payload.indexes.includes(i)
+          ? {
+              ...m,
+              visibility: action.payload.newVisibility,
+              clicked: action.payload.newVisibility,
+            }
+          : m;
+      }),
+    },
+  };
+}
+
+function onSwitchVSBYToggle(
```

Comment:
no reader will know what this acronym means, make it like `onAnswerVisibilityToggle`